.github/workflows/ci.yml

```diff
@@ -0,0 +1,327 @@
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: ${{ env.key }}-plan-${{ hashFiles('dist-newstyle/cache/plan.json') }}
+        restore-keys: ${{ env.key }}-
+
+    - name: 🛠️ Build Cabal dependencies
+      run: |
+        cabal build all \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}" \
+          --only-dependencies
+
+    - name: 💾 Save Cabal dependencies
+      uses: actions/cache/save@v4
+      if: ${{ !env.ACT && steps.cache-cabal.outputs.cache-hit != 'true' }}
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: ${{ steps.cache-cabal.outputs.cache-primary-key }}
+
+    - name: 🏗️ Build
+      if: ${{ matrix.cabal-tests }}
+      run: |
+        cabal build all \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}"
+
+    - name: 🧪 Test
+      if: ${{ matrix.cabal-tests }}
+      run: |
+        cabal test all \
+          -j1 \
+          --test-show-details=direct \
+          --project-file="${{ matrix.cabal-project-file || env.DEFAULT_CABAL_PROJECT_FILE }}"
+      env:
+        TASTY_TIMEOUT: "5m"
+
+    - name: 🛠️ Setup cabal-docspec (Linux)
+      if: ${{ matrix.cabal-tests && runner.os == 'Linux' }}
+      uses: ./.github/actions/setup-cabal-docspec
+
+    - name: 🧪 Test with cabal-docspec (Linux)
+      if: ${{ matrix.cabal-tests && runner.os == 'Linux' }}
+      run: ./scripts/test-cabal-docspec.sh
+      env:
+        SKIP_CABAL_BUILD: true
+
+    - name: 🏗️ Build documentation
+      if: ${{ matrix.cabal-documentation }}
+      run: cabal haddock-project
+
+    - name: 📦 Upload documentation
+      if: ${{ matrix.cabal-documentation }}
+      uses: actions/upload-artifact@v4
+      with:
+        name: haddocks-${{ runner.os }}-ghc-${{ matrix.ghc-version || env.DEFAULT_GHC_VERSION }}-cabal-${{ matrix.cabal-version || env.DEFAULT_CABAL_VERSION }}
+        path: haddocks
+        if-no-files-found: error
+        retention-days: 1
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macOS-latest", "windows-latest"]
+        ghc-version: ["8.10", "9.2", "9.4", "9.6", "9.8", "9.10", "9.12"]
+        cabal-version: ["3.10"]
+        # The inclusion of these default keys in the matrix ensures that any
+        # entry under include that assigns these keys creates a new matrix entry
+        cabal-flags: [""]
+        cabal-project-file: [""]
+        cabal-documentation: [false]
+      
+        exclude:
+          # GHC 8.10 is not supported on macOS wit ARM CPU
+          - ghc-version: "8.10"
+            os: "macOS-latest"
+
+        include:
+          # Include builds for various special cases
+          - name: "Build on macOS with Intel CPU and GHC 8.10"
+            os: "macOS-13"
+            ghc-version: "8.10"
```

Comment:
But here I'm seeing
```
{
  os: macOS-13,
  ghc-version: 8.10,
  name: Build on macOS with Intel CPU and GHC 8.10
}
```
in the output, so `cabal-tests`, `cabal-benchmarks` and `cabal-documentation` are unexpectedly empty

Reply:
I misremembered the semantics of `include`, which only modifies original entries. I've fixed this by relying on the absence, and renaming `cabal-tests` to `cabal-skip-tests` and likewise for the benchmarks.